assembly/src/assembler/mast_forest_builder.rs

```diff
@@ -456,14 +456,18 @@ impl MastForestBuilder {
     }
 
     pub fn set_before_enter(&mut self, node_id: MastNodeId, decorator_ids: Vec<DecoratorId>) {
-        self.mast_forest[node_id].set_before_enter(decorator_ids);
+        // Does this need to have a Vec<DecoratorSpan>?
+        let span = DecoratorSpan::new_collection(decorator_ids).into_iter().next().unwrap();
```

Comment:
Not sure this is needed, but `new_collection` method creates a collection because it seems that it may be possible to have non-contiguous decorator spans. See this comment: https://github.com/0xPolygonMiden/miden-vm/pull/1621/files#r1912588483